Source/Core/CesiumTerrainProvider.js

```diff
@@ -828,6 +969,40 @@ define([
         return this._levelZeroMaximumGeometricError / (1 << level);
     };
 
+    var scratchTileXY = new Cartesian2();
+    /**
+     * Loads the availability for the position if not already loaded.
+     *
+     * @param {Cartographic} position The position to make sure availability is loaded for.
+     * @returns {undefined|Promise} A promise that resolves when the availability is loaded. Undefined if availability isn't supported.
+     */
+    CesiumTerrainProvider.prototype.loadAvailability = function(position) {
+        if (!defined(this._availability)) {
+            return undefined;
+        }
+
+        // Try and load the max level tile at the position. Even if its not there we will
+        //  still end up loading all the BVH tiles below it and its max level of availability.
+        var level = this.availability._maximumLevel;
+        this._tilingScheme.positionToTileXY(position, level, scratchTileXY);
+
+        return when(loadTileAvailability(this, scratchTileXY.x, scratchTileXY.y, level));
```

Comment:
So every time you ask the terrain provider to load availability it will create a promise, and then, if availability isn't yet know for the maximum level at that position, it will do a non-throttled request for the closest BVH node to that level (generating 404s if it doesn't exist). I modifed the Terrain sandcastle example to use your Zion terrain and sample heights around Zion. Pressing the "Sample Most Detailed Everest Terrain" button generates thousands of requests all at once and locks up the browser. Here's the [Super Hacky Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=xVlbd9pIEv4rvXlBPsGN7pc4yVkbOwkZYyc2k0wS5+xppAY6FmoiNQY8x2/7z/aPbbVuSCA8ZDOzowcD6rpXddXX7S5N2HyKezzCAR2ReSiOfZ8myYDf0gi9QC26ejsZvvbZJXvb+/W+p12wXtKLriy/27N7t7PfPnTfehiIvgevbyXR7afp2+nnU3/ZX2ms/+1scT54v+pPP4nL1z3jsqtNLwZXk4vTANbGy/60z867b2efQVh/tWDk4yu1940vLwY9vX/6q3Vxf7wYvcf+qPv+uzicmJ8DU5B//Xb27V54ztxwu+TSu1suFyPn+v2J2xsGvdbRTdTd8umaxnc0lu5MhJglzzqdgN4dkhnDfkbr86nkvInuSIzuGY8GNI4JkxGI6ALlErOPfOldzO9YQGPl95sIwTOPw2dorfqKJnwe+xSPYj49ThIqeoFiOwc30cNBqemO0UVqV0XJh/Sd0sos6/JIgDIat9ooVyTq+tGzqsGl+E4HgU4UsvFEsGiMBEcinlPQm8rHiU8jischH1JMIzIM6XlB+iKlrEiZzxAJQW1EBLujhQFolluQZM7QMGSzhLOgtKzm11mxvBm/SjjiwYR+5HG4Q8SHq2J9dw4gHGmSIceLxQLfxYdiQheSRya5U/ndESykiYZVrHYco9NqZzL8mAZMSDFFDQREEORDNgVNVujDAPX/8+9fWmWkr0kU+CQRIcUkCAach0MS92k0V76UGVumAhvrBx3m7qHUM5QvF+bwKKEh9SX/aB75AlKtHBSlIJ88n5tV8QItKoE62qLfJ/+S/AH8rJTejzsCryOO6GgETiQ/71VeCpAlImq1kNbRn2PxooOKYPyVBle45RPT73OaiA8U6mx5weMpCRNQl+3agujh4O9MJQTmI3gR//+jkqrtk+R2d0Qa3dvVdX7eg61295gZOzvXz9ux3TQrhnxto1bOIjtSq6ljnWYz8rHG1efzSKAzmKGQir1NDjm/PRZ9kfM9vkHfkHCETvmU7h8RmBiCxGOYcvUpTWSjZiQylEPddD1bt7DqGrauW5rptNGhaXiGZ2rYhMdQdbONDNfVLcPCnmmZmmO4nlPbZqCIj0bJY4ps13KwqXqqrhqOZdptZGo6NnTXdA3L9kzDcttIB/3YskCJbruarupGw272yZTGBGfBUzIH27n+PyIfxCRKRtA6lNzEPhExW5q4d3oGmGrw6dEUQFmgVyDBZ4nP0QlZ/bmpcFTX1UzsWpZrQF5kInQIt+Vh01Jd0/JUR2bC00zVkJlwTN2Djx/MhKPapo5tW4XMqp5pW6DH0Gxbw5bmeJANzXFTRZalWhheeZ6jmYZh2H9nLuQ+ved8ejIXAqLdWmOifPdek+kspO94wmT0k3S5yMX2Ritys0deDFV1HBVDBhwdgm+4sB0s2zA9KF/N0GCjWC7EUPccV7N0rLu2LGBNK8O1R05AnIk9B7aHazuuZeuQE8czsGFqnuO5qqs5pgY6TM/DHlji2qpjQ9L0Usf+6fifU/FQC2mSxjtv1tfz9EykNOeiFu1yKqxnXDmC8MfX166Zpk6SdzonK5QfUNrFvA04TQAvCcSHyTymaEyhJYp4VbAM6YRFQVEVGF1TkcH7CUvQKCRjlCGABN4QgWsxqeKEgM7EZADFcjwGOYlYn3dK7F/hpJEAX2mCk3kyo1EAhRaJRNmIeUkV0ym/o8dhmGN7SQPhR4qMEAMd6hF8PN9R2zik0VhMjtDTp2yry8xyKmloI/cX9rXU2WQbDLxNiBFBrUBrK0TjCZXwCQv+ii1poGgH7Tp9acOzdbqh4GLBxzGZTZg/4GXxKwXxppAhC8MhJ3EAUjbsSc0GFMh8El7GbMykpuIIVHuPTy4Hg8t+e5s/ARrplIqdhlU2JWO52sK4k35POiPis5CJFR6zUavO8bAhISRDGjabnY+SvUOZVUaUjh9NnQk05RFPZsQvYUD1mfCY3csD8VZY3mys4C5s7LOrBhkzMCS8zPrVs8Z+pSsq9CfNbLQVstblIY/XqtOf+OT8uPtLAwOfixCO7z/Ek0z44oT4t+MYYFeQA94GumFJU4ivuiNfKSr22qj6xzl4VNA72B6yneyKjMRKBxvV0QzFm/tCMp/SSvOo99zsMPA6ZoESA9og0TikEhZes3taa7JjIPnIAjGBLmBqR/WFN2nJVVc6nbQzZ8PxnIAx84Cu+zMMgwlU6BUJwMVE0R3sSThYQsA6O4/Gj/O7NvYk0rQ2+OvT8apwUNkSfIi2vG9vWf8oUSnq6T6iGojWbbswfdPs9FqrcPlQg+muOzpACQ1gl2aaelvFtukA+nVVw3A8B8BcOyUDEt2wdMeyAYgZKRlADEsGzJYwza3rbm7yYNCXr0cbg2WVDZYVDJZ1HcgpsqpNkZJ+mdEvc/q0oCT58mCzs0nqcEfiQxrPFIoXEFEILaZEfi5RBynrKj1E2kEVLpYym2sxF5lA75hImRGP5ZdVITSv8F1SKyNyYxMX80kpnWmXNmxK2jGbZ/Nksp5pFZ6H2s6PqZjH0W7w+tB0cTaGusoAsNI6S1FM5Rrmjy872uvDiT+h/i0N1k1jj4uSnOdo571eg3kjvmUZvMplB/saVGHZx470mF4YkoW2OJqXFzVEoHPY5yHyWu2GM9ujO6vShlWsqmUfywtpMaFRgaNrKFlpvqJoIxg8O5BzuxFnFwqb7g/g1NrKeOpHpX2C1OcQoVMqCJOR3ghZY5zYCCn/yH0FpA6DPNjhJSZ3IJYMUwx1UOsgPljJwa6Qj5VWzV9pUGkPy3B/Mp/NYL/DC9moxISi7OxNg61791Z1/2U7bmP+/kCa989z1eqdOf/LE/6k/eR5IlYhfZnx/pNNZdzkvwAUQLWCAgv4l3SGc9hOAvuZmuedgul5wO4QC17cPNn4T8vNE+SHJElgZTQPQzkPb568fN4B+hpbyInESpdgbEhWkmSivTzPXmKMn3fg5zaXyEoTyGtv13d0jboqdwON67UIVSnyz/8C).